escrow/lib.rs

```diff
@@ -0,0 +1,260 @@
+        use ink_lang as ink;
+
+        fn get_default_test_accounts() -> ink_env::test::DefaultAccounts<ink_env::DefaultEnvironment>
+        {
+            ink_env::test::default_accounts::<ink_env::DefaultEnvironment>()
+        }
+
+        fn set_balance(account_id: AccountId, balance: Balance) {
+            ink_env::test::set_account_balance::<ink_env::DefaultEnvironment>(account_id, balance)
+        }
+
+        fn get_balance(account_id: AccountId) -> Balance {
+            ink_env::test::get_account_balance::<ink_env::DefaultEnvironment>(account_id)
+                .expect("Cannot get account balance")
+        }
+
+        fn contract_id() -> AccountId {
+            ink_env::test::callee::<ink_env::DefaultEnvironment>()
+        }
+
+        fn set_caller(caller: AccountId) {
+            ink_env::test::set_caller::<ink_env::DefaultEnvironment>(caller);
+        }
+
+        fn create_contract(deposit: Balance) -> (Escrow, AccountId, AccountId) {
+            let accounts = get_default_test_accounts();
+            let buyer = accounts.eve;
+            let seller = accounts.frank;
+
+            set_caller(buyer);
+            set_balance(contract_id(), deposit);
+            let escrow = Escrow::new(seller);
+
+            (escrow, buyer, seller)
+        }
+
+        fn confirm(escrow: &mut Escrow, buyer: AccountId) -> () {
+            set_caller(buyer);
+            escrow.confirm().expect("Confirm failed!")
+        }
+
+        fn refund(escrow: &mut Escrow, seller: AccountId) -> () {
+            set_caller(seller);
+            escrow.refund().expect("Refund failed!")
+        }
+
+        #[ink::test]
+        fn when_constructor_is_called_then_contract_has_deposit() {
+            const DEPOSIT: Balance = 123;
+            let (escrow, buyer, seller) = create_contract(DEPOSIT);
+            assert_eq!(escrow.buyer, buyer);
+            assert_eq!(escrow.seller, seller);
+            assert_eq!(get_balance(contract_id()), DEPOSIT);
+        }
+
+        #[ink::test]
+        fn when_buyer_confirms_then_seller_receives_deposit() {
+            const DEPOSIT: Balance = 10;
+            let (mut escrow, buyer, _) = create_contract(DEPOSIT);
+
+            let should_terminate = move || confirm(&mut escrow, buyer);
+            // unfortunately we can't test much in this UT, since below test only outcome of
+            // self.env().terminate_contract(); the confirm() panics in off-chain testing environment
+            // so there's no easy way to test state directly of the escrow contract after confirm()
+            ink_env::test::assert_contract_termination::<ink_env::DefaultEnvironment, _>(
+                should_terminate,
+                buyer,
+                DEPOSIT,
+            );
+            // we can only indirectly check account balances
+            assert_eq!(get_balance(contract_id()), 0);
+            // below incorrectly returns account values (should be other way round)
```

Comment:
?

Reply:
Since if I uncomment below lines
```
// assert_eq!(get_balance(seller), DEPOSIT);
// assert_eq!(get_balance(buyer), 0);
```
they will be true in the **current** code version - and this is incorrect when the refunding, the buyer should have the deposit back, and not the seller. This happens because of this issue https://github.com/paritytech/ink/issues/1117